runtime/runtime/src/actions.rs

```diff
@@ -789,6 +848,73 @@ pub(crate) fn action_add_key(
     Ok(())
 }
 
+pub(crate) fn action_add_gas_key(
+    apply_state: &ApplyState,
+    state_update: &mut TrieUpdate,
+    account: &mut Account,
+    result: &mut ActionResult,
+    account_id: &AccountId,
+    add_gas_key: &AddGasKeyAction,
+) -> Result<(), StorageError> {
+    if get_gas_key(state_update, account_id, &add_gas_key.public_key)?.is_some() {
```

Comment:
Will we later add an additional check to make sure that access keys and gas keys don't share public keys or are we going to allow having the same public key for both gas key and access key?